ui/components/app/confirm-page-container/confirm-page-container.component.js

```diff
@@ -359,6 +381,23 @@ const ConfirmPageContainer = (props) => {
             <AdvancedGasFeePopover />
           </>
         )}
+        {
+          ///: BEGIN:ONLY_INCLUDE_IN(build-flask)
+        }
+        {insightObject?.warnings?.length > 0 && isShowingTxInsightWarnings && (
```

Comment:
Isn't the `insightObject?.warnings?.length > 0` redundant since you're already checking this in here:

``` const topLevelHandleSubmit = () => {
    ///: BEGIN:ONLY_INCLUDE_IN(build-flask)
    if (insightObject?.warnings?.length > 0) {
      return setIsShowingTxInsightWarnings(true);
    }
    ///: END:ONLY_INCLUDE_IN
    return handleSubmit();
  };```

Reply:
good catch